packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -48,75 +49,159 @@ class _ProductListPageState extends State<ProductListPage> {
     }
     return Scaffold(
       appBar: AppBar(
-        title: Row(
-          children: <Widget>[
-            Flexible(
-              child: Text(
-                ProductQueryPageHelper.getProductListLabel(
-                  productList,
-                  context,
-                  verbose: false,
-                ),
+        title: _selectionMode
+            ? Text(
+                '${_selectedBarcodes.length}/${products.length} products', // TODO(monsieurtanuki): localize
                 overflow: TextOverflow.fade,
+              )
+            : Row(
+                children: <Widget>[
+                  Flexible(
+                    child: Text(
+                      ProductQueryPageHelper.getProductListLabel(
+                        productList,
+                        context,
+                        verbose: false,
+                      ),
+                      overflow: TextOverflow.fade,
+                    ),
+                  ),
+                ],
               ),
+        actions: <Widget>[
+          if (_selectionMode && _selectedBarcodes.isNotEmpty)
+            IconButton(
+              icon: const Icon(Icons.check_box_outline_blank),
+              onPressed: () => setState(() => _selectedBarcodes.clear()),
             ),
-          ],
-        ),
-        actions: !dismissible
-            ? null
-            : <Widget>[
-                PopupMenuButton<String>(
-                  itemBuilder: (final BuildContext context) =>
-                      <PopupMenuEntry<String>>[
-                    PopupMenuItem<String>(
-                      value: 'clear',
-                      child: Text(appLocalizations.clear),
-                      enabled: true,
-                    ),
-                  ],
-                  onSelected: (final String value) async {
-                    switch (value) {
-                      case 'clear':
-                        if (await ProductListDialogHelper.instance
-                            .openClear(context, daoProductList, productList)) {
-                          localDatabase.notifyListeners();
-                        }
-                        break;
-                      default:
-                        throw Exception('Unknown value: $value');
-                    }
-                  },
+          if (_selectionMode && _selectedBarcodes.length < products.length)
+            IconButton(
+              icon: const Icon(Icons.check_box),
+              onPressed: () => setState(() => _populateAll(products)),
+            ),
+          if (dismissible && !_selectionMode)
+            PopupMenuButton<String>(
+              itemBuilder: (final BuildContext context) =>
+                  <PopupMenuEntry<String>>[
```

Comment:
instead of String, can we use an Enum here?

Then the switch case can be exhaustive, no need for default: throw exception